sno/working_copy/base.py

```diff
@@ -381,22 +381,92 @@ def _execute_dirty_pks_query(self, db, dataset):
         Queries the tracking table for the rows belonging to the given dataset, and returns a result
         containing all the primary keys of all the rows that have been inserted / updated / deleted.
         """
-        raise NotImplementedError()
+        return db.execute(
+            f"""SELECT pk FROM {self.TRACKING_TABLE} WHERE table_name = :table_name;""",
+            {"table_name": dataset.table_name},
+        )
 
-    def _execute_dirty_rows_query(self, db, dataset, feature_filter):
+    def _execute_dirty_rows_query(
+        self, db, dataset, feature_filter=None, meta_diff=None
+    ):
         """
         Does a join on the tracking table and the table for the given dataset, and returns a result
         containing all the rows that have been inserted / updated / deleted.
         """
-        raise NotImplementedError()
+        # TODO - use sqlalchemy, move this if-block.
+        if hasattr(self, "schema"):
+            table_identifier = f'"{self.schema}"."{dataset.table_name}"'
+            cast = "::text"
+        else:
+            table_identifier = f'"{dataset.table_name}"'
+            cast = ""
+
+        feature_filter = feature_filter or UNFILTERED
+
+        if (
+            meta_diff
+            and "schema.json" in meta_diff
+            and meta_diff["schema.json"].new_value
+        ):
+            schema = Schema.from_column_dicts(meta_diff["schema.json"].new_value)
+        else:
+            schema = dataset.schema
+
+        pk_field = schema.pk_columns[0].name
+        col_names = ",".join([f'TAB."{col.name}"' for col in schema])
+
+        diff_sql = f"""
+            SELECT
+                TRA.pk AS ".__track_pk",
+                {col_names}
+            FROM {self.TRACKING_TABLE} TRA LEFT OUTER JOIN {table_identifier} TAB
```

Comment:
I would imagine the local variable would have precedence over a global table. Will have to check, since we are using these elsewhere...

This is a good candidate for using a sqlalchemy select() function and let it build the query itself. Will probably do that, then we don't need aliases anyway.